samtranslator/model/eventsources/push.py

```diff
@@ -921,7 +926,39 @@ def _add_swagger_integration(  # type: ignore[no-untyped-def] # noqa: too-many-s
                 path=self.Path, method_name=self.Method, request_parameters=parameters
             )
 
-        api["DefinitionBody"] = editor.swagger
+        if merge_definitions:
+            api["DefinitionBody"] = self._get_merged_definitions(api["DefinitionBody"], editor.swagger)
+        else:
+            api["DefinitionBody"] = editor.swagger
+
+    def _get_merged_definitions(
+        self, source_definition_body: Dict[str, Any], dest_definition_body: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Merge SAM generated swagger definition(dest_definition_body) into inline DefinitionBody(source_definition_body):
+        - for a conflicting key, use SAM generated value
+        - otherwise include key-value pairs from both definitions
+        """
+        merged_definition_body = source_definition_body.copy()
+        source_body_paths = merged_definition_body.get("paths", {})
+        if (
+            not isinstance(source_body_paths, dict)
+            or not isinstance(source_body_paths.get(self.Path, {}), dict)
+            or not isinstance(source_body_paths.get(self.Path, {}).get(self.Method, {}), dict)
+        ):
+            # Do nothing if swagger body is invalid. We are already raising exception later by validating swagger body
```

Comment:
we can still raise exceptions here. Relying on knowledge like this is not reliable. Imagine in the future that validation mentioned changes behavior this comment will be outdated.